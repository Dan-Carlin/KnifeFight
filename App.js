import { useState } from "react";
import { StatusBar } from "expo-status-bar";
import { View } from "react-native";
import AppLoading from "expo-app-loading";

import KFHomeScreen from "./app/screens/home/KFHomeScreen";
import KFSetupStepOneScreen from "./app/screens/setup/KFSetupStepOneScreen";
import KFSetupStepTwoScreen from "./app/screens/setup/KFSetupStepTwoScreen";
import KFSetupStepThreeScreen from "./app/screens/setup/KFSetupStepThreeScreen";
import KFBannerScreen from "./app/screens/tools/KFBannerScreen";
import KFToolsScreen from "./app/screens/tools/KFToolsScreen";
import useFonts from "./app/hooks/useFonts";

export default function App() {
  const [fontsLoaded, setFontsLoaded] = useState(false);

  const LoadFonts = async () => {
    await useFonts();
  };

  if (!fontsLoaded) {
    return (
      <AppLoading
        startAsync={LoadFonts}
        onFinish={() => setFontsLoaded(true)}
        onError={() => {}}
      />
    );
  }

  return (
    <>
      <StatusBar style="light" backgroundColor="transparent" />
      <KFToolsScreen />
    </>
  );
}
