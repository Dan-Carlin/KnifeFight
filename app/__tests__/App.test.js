import React from "react";
import renderer from "react-test-renderer";

import App from "../../App";

describe("App tests", () => {
  it("renders App correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
