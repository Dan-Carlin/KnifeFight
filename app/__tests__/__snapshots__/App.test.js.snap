// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders App correctly 1`] = `null`;
